### PR TITLE
upgraded network module

### PR DESCRIPTION
**Pull request by nakonn**

### CODE REVIEW

File: terraform/02-variables.tf [hunk 1]
```diff
@@ -0,0 +1,17 @@
+variable "vpc_cidr" {  
+  default     = "10.0.0.0/16"
+  type        = string
+  description = "CIDR block for VPC"
```
Comment (Jiraiya106) [inline terraform/02-variables.tf:4]: validation

File: terraform/01-main.tf [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  required_version = ">= 0.13" # min Terraform version
+}
+
+provider "aws" {
+  region = var.region
+}
+
+module "network" {  
+  source = "git@github.com:dvsp-itransition/terraform-modules.git//simple_network"
```
Comment (Jiraiya106) [inline terraform/01-main.tf:17]: he turns into himself?

File: terraform/modules/simple_network/4-routes.tf [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# Public Route Tables
+resource "aws_route_table" "public_route_tables" {
+  for_each = aws_subnet.public_subnets
+  vpc_id = aws_vpc.this.id
+  
+  route {
```
Comment (Jiraiya106) [inline terraform/modules/simple_network/4-routes.tf:6]: dynamic block

File: terraform/modules/simple_network/4-routes.tf [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    cidr_block = "0.0.0.0/0"
```
Comment (Jiraiya106) [inline terraform/modules/simple_network/4-routes.tf:7]: variable

File: terraform/modules/simple_network/4-routes.tf [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+# Private route tables
+resource "aws_route_table" "private_route_tables" {
+  for_each = aws_subnet.private_subnets
+  vpc_id = aws_vpc.this.id
+  depends_on = [ aws_eip.nat_gateway_eip ]
+
+  route {
+    cidr_block = "0.0.0.0/0"
```
Comment (Jiraiya106) [inline terraform/modules/simple_network/4-routes.tf:30]: variable

File: terraform/modules/simple_network/4-routes.tf [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  route_table_id = aws_route_table.public_route_tables[each.key].id    
+}
+
+# Private route tables
+resource "aws_route_table" "private_route_tables" {
+  for_each = aws_subnet.private_subnets
+  vpc_id = aws_vpc.this.id
+  depends_on = [ aws_eip.nat_gateway_eip ]
+
+  route {
```
Comment (Jiraiya106) [inline terraform/modules/simple_network/4-routes.tf:29]: dynamic block

File: terraform/01-main.tf (hunk 2 above)
Comment (nakonn) [reply terraform/01-main.tf:17]: Nope, this is separate repository

File: terraform/02-variables.tf (hunk 1 above)
Comment (nakonn) [reply terraform/02-variables.tf:4]: configured

File: terraform/modules/simple_network/4-routes.tf (hunk 3 above)
Comment (nakonn) [reply terraform/modules/simple_network/4-routes.tf:6]: corrected

File: terraform/modules/simple_network/4-routes.tf (hunk 4 above)
Comment (nakonn) [reply terraform/modules/simple_network/4-routes.tf:7]: fixed

File: terraform/modules/simple_network/4-routes.tf (hunk 5 above)
Comment (nakonn) [reply terraform/modules/simple_network/4-routes.tf:30]: fixed

File: terraform/modules/simple_network/4-routes.tf (hunk 6 above)
Comment (nakonn) [reply terraform/modules/simple_network/4-routes.tf:29]: configured

**Pull request closed without merging**